wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceInfoViewModel.swift

```diff
@@ -0,0 +1,142 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import SwiftUI
+import WireCommonComponents
+
+struct E2EIdentityCertificate {
+    var status: E2EIdentityCertificateStatus
+    var serialNumber: String
+    var certificate: String = .randomString(length: 500)
+    var exipirationDate: Date = .now
+    var isExpiringSoon: Bool  = false
+}
+
+enum E2EIdentityCertificateStatus: CaseIterable {
+    case notActivated, revoked, expired, valid, none
+}
+
+extension E2EIdentityCertificateStatus {
+    func titleForStatus() -> String {
+        switch self {
+        case .notActivated:
+            return "device.details.not.activated".localized
+        case .revoked:
+            return "device.details.revoked".localized
+        case .expired:
+            return "device.details.expired".localized
+        case .valid:
+            return "device.details.valid".localized
+        case .none:
+            return ""
+        }
+    }
+
+    func imageForStatus() -> Image {
+        switch self {
+        case .notActivated:
+            return Image(.certificateExpired)
+        case .revoked:
+            return Image(.certificateRevoked)
+        case .expired:
+            return Image(.certificateExpired)
+        case .valid:
+            return Image(.certificateValid)
+        case .none:
+            return Image(asset: .init(name: ""))
+        }
+    }
+}
+
+protocol DeviceDetailsViewActions {
+    func fetchCertificate() async
+    func showCertificate(validate: () -> Bool, result: (Bool) -> Void)
+    func removeDevice()
+    func resetSession()
+    func setVerified(_ result: (Bool) -> Void)
+    func copyToClipboard(_ value: String)
+}
+
+final class DeviceDetailsActionsHandler: DeviceDetailsViewActions {
+    var isDeviceVerified: Bool = false
+
+    func fetchCertificate() async {
+
+    }
+
+    func showCertificate(validate: () -> Bool, result: (Bool) -> Void) {
```

Comment:
could you explain what the two blocks will do, not sure why we have it expecially the result one?

Reply:
The first one takes inputs for validations and the second one gives us an instruction whether the certificate is ready to be shown or not. Open to change this based on the implementation logic later by @KaterinaWire

Reply:
simplified this